File: 백준/Silver/10845. 큐/큐.py
from collections import deque
import sys
input = sys.stdin.readline
class Queue:

    def __init__(self):
        self.queue= deque()
    def push(self, x):
        self.queue.append(x)
    def pop(self):
        if self.empty():
            return -1
        else:
            return self.queue.popleft()
    def size(self):
        return len(self.queue)
    def empty(self):
        if self.queue:
            return 0
        else: 
            return 1
    def front(self):
        if(self.empty()): return -1
        else: return self.queue[0]
    def back(self):
        if(self.empty()): return -1
        else: return self.queue[-1]

queue = Queue()
n = int(input())
result =[]
for _ in range(n):
    cmd = input().split()
    if(cmd[0]== 'push'):
        queue.push(cmd[1])
    elif(cmd[0] == 'pop'):
        result.append(queue.pop())
    elif(cmd[0] == 'size'):
        result.append(queue.size())
    elif(cmd[0] == 'empty'):
        result.append(queue.empty())
    elif(cmd[0] == 'front'):
        result.append(queue.front())
    elif(cmd[0] == 'back'):
        result.append(queue.back())

print('\n'.join(map(str, result)))

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader((System.in)));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Queue<Integer> q = new LinkedList<>();
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int lastValue = -1;
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();
            switch (cmd) {
                case "push":
                    lastValue = Integer.parseInt(st.nextToken());
                    q.offer(lastValue);
                    break;
                case "pop":
                    bw.write(q.isEmpty() ? "-1\n" : q.poll() + "\n");
                    break;
                case "size":
                    bw.write(q.size() + "\n");
                    break;
                case "empty":
                    bw.write(q.isEmpty() ? "1\n" : "0\n");
                    break;
                case "front":
                    bw.write(q.isEmpty() ? "-1\n" :q.peek() + "\n");
                    break;
                case "back":
                    bw.write(q.isEmpty() ? "-1\n" : lastValue + "\n");
                    break;
            }
        }
        bw.flush();
        bw.close();

    }
}